Unichain Mainnet Petra L1 Upgrade (#713)
ethereum-optimism/superchain-ops: tasks/eth/unichain-001-petra-l1-upgrade/VALIDATION.md

# Validation

This document can be used to validate the state diff resulting from the execution of the upgrade transaction.

For each contract listed in the state diff, please verify that no contracts or state changes shown in the Tenderly diff
are missing from this document. Additionally, please verify that for each contract:

- The following state changes (and none others) are made to that contract. This validates that no unexpected state
  changes occur.
- All addresses (in section headers and storage values) match the provided name, using the Etherscan and Superchain
  Registry links provided. This validates the bytecode deployed at the addresses contains the correct logic.
- All key values match the semantic meaning provided, which can be validated using the storage layout links provided.

> 

Reviewer: The expected Domain Hash is missing and now should be required. 

```
## Expected Domain and Message Hashes

> [!CAUTION]
>
> Before signing, ensure the below hashes match what is on your ledger.
>
> ### Security Council
>
> - Domain Hash: ``
> - Message Hash: ``
>
> ### Optimism Foundation
>
> - Domain Hash: ``
> - Message Hash: ``
```

> ## State Changes

Reviewer: We are missing the override here, this is important since we will sign with not the same nonce onchain.

> Note: The changes listed below do not include safe nonce updates or liveness guard related changes.

Reviewer: One override is required still because we are currently signing the task with not the correct nonce for now since 2 tasks will be executed before. 
the value are not correct and will double check after finishing to modifying the validation.md
```
 - U3 (0x6d5B183F538ABB8572F5cD17109c617b994D5833)
Set the nonce to expect value of the execution by setting the nonce to `1`:

Key: 0x0000000000000000000000000000000000000000000000000000000000000004
Value: 0x0000000000000000000000000000000000000000000000000000000000000001
Meaning: The nonce is overwrite to nonce env file of the .****
```

@winnsterx

You can verify this absolute prestate by running the following [command](https://github.com/ethereum-optimism/optimism/blob/6819d8a4e787df2adcd09305bc3057e2ca4e58d9/Makefile#L133-L135) in the root of the monorepo:

```bash
make reproducible-prestate
```

You should expect the following output at the end of the command:

```bash
Cannon Absolute prestate hash: 
0x035ac388b5cb22acf52a2063cfde108d09b1888655d21f02f595f9c3ea6cbdcd
Cannon64 Absolute prestate hash: 
0x03a7d967025dc434a9ca65154acdb88a7b658147b9b049f0b2f5ecfb9179b0fe
CannonInterop Absolute prestate hash: 
0x0379d61de1833af6766f07b4ed931d85b3f6282508bbcbf9f4637398d97b61c1
```
## State Overrides
Note: The changes listed below do not include threshold and number of owners overrides or liveness guard related changes, these changes are listed in the [NESTED-VALIDATION.md](../../../NESTED-VALIDATION.md) file.

### `0xc2819DC788505Aac350142A7A707BF9D03E3Bd03` (Council Safe)

> We override the nonce of Security Council `OwnerSafe` contract.

Reviewer: ```suggestion
We override the nonce of `Security Council`. 
```

- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000005` <br/>
  **Value:** `0x0000000000000000000000000000000000000000000000000000000000000014` <br/>

>   **Meaning:** Override the nonce value of the Security Council's `OwnerSafe` by increasing from 18 to 20.

Reviewer: ```suggestion
  **Meaning:** Override the nonce value of the `Security Council` by increasing from 18 to 20.
```

### `0x847B5c174615B1B7fDF770882256e2D3E95b9D92` (Foundation Upgrade Safe)
- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000005` <br/>
  **Value:** `0x0000000000000000000000000000000000000000000000000000000000000012` <br/>

>   **Meaning:** Override the nonce value of the Security Council's `OwnerSafe` by increasing from 16 to 18.

Reviewer: ```suggestion
  **Meaning:** Override the nonce value of the Foundation Upgrade Safe by increasing from 16 to 18.
```

For details on other state overrides, please refer to the [NESTED-VALIDATION.md](../../../NESTED-VALIDATION.md) file.

> * Liveness guard related changes are listed [here](../../../NESTED-VALIDATION.md#liveness-guard-security-council-safe-or-unichain-operation-safe-only) file.

Reviewer: ```suggestion
```

## State Changes

### `0x2F12d621a16e2d3285929C9996f478508951dFe4` (`DisputeGameFactoryProxy`)

- **Key**: `0xffdfc1249c027f9191656349feb0761381bb32c9f557e01f419fd08754bf5a1b` <br/>
  **Before**: `0x00000000000000000000000008f0f8f4e792d21e16289db7a80759323c446f61` <br/>
  **After**: `0x000000000000000000000000171dE4f3ea4fBbf233aA9649dCf1b1d6fD70f542` <br/>
  **Meaning**: Updates the implementation for game type 0. Verify that the new implementation is set using
  `cast call 0x2F12d621a16e2d3285929C9996f478508951dFe4 "gameImpls(uint32)(address)" 0`.

### `0x2F12d621a16e2d3285929C9996f478508951dFe4` (`DisputeGameFactoryProxy`)

- **Key**: `0x4d5a9bd2e41301728d41c8e705190becb4e74abe869f75bdb405b63716a35f9e` <br/>
  **Before**: `0x000000000000000000000000c457172937ffa9306099ec4f2317903254bf7223` <br/>
  **After**: `0x0000000000000000000000003aFdc7cCF8a1c0d351E3E5F220AF056ea2c07733` <br/>
  **Meaning**: Updates the implementation for game type 1. Verify that the new implementation is set using
  `cast call 0x2F12d621a16e2d3285929C9996f478508951dFe4 "gameImpls(uint32)(address)" 1`.

> 

Reviewer: ```suggestion
### Liveness Guards

Liveness Guard related changes are listed [here](../../../NESTED-VALIDATION.md#liveness-guard-security-council-safe-or-unichain-operation-safe-only) file.
```
